core/frontend/src/DisplayStyleState.ts

```diff
@@ -409,20 +409,33 @@ export abstract class DisplayStyleState extends ElementState implements DisplayS
   }
 
   /** @internal */
-  public changeBaseMapProps(props: MapLayerProps | ColorDef) {
+  public changeBaseMapProps(props: Partial<MapLayerProps>| ColorDef): boolean {
```

Comment:
Because Im now passing `Partial<MapLayerProps>` (to support calls like `changeBaseMapProps({visible:false})`, I can no longer call `BaseMapLayerSettings.fromJSON`, which require now require formatId,name,url.

Problem is, if `this.settings.mapImagery.backgroundBase` is a color, then we can no longer 'auto switch' to MapLayerSettings.

Reply:
So, get rid of changeBaseMapProps - it is an awkward and unhelpful API, and it's marked `@internal`. Let caller figure out what they want the base layer to be and set it to MapImagerySettings.backgroundBase.
e.g.,
```
if (mapImagery.backgroundBase instanceof BaseMapLayerSettings)
  mapImagery.backgroundBase = mapImagery.backgroundBase.clone({ some subset of properties to change });
else
  mapImagery.backgroundBase = BaseMapLayerSettings.fromJSON({ properties of new base layer });
```

Reply:
Although I suppose that won't work properly until we fix the need to manually synchronize the MapImagerySettings with the DisplayStyleSettings' JSON every time it changes...